libs/blocks/merch-card/merch-card.js

```diff
@@ -112,103 +203,156 @@ const decorateMerchCardLinkAnalytics = (el) => {
     if (ribbonMetadata !== null) {
-      const badge = returnRibbonStyle(ribbonMetadata);
+      const badge = returnRibbonStyle(ribbonMetadata.children);
       if (badge !== null) {
-        merchCard.setAttribute('badge-background-color', badge.badgeBackgroundColor);
+        merchCard.setAttribute(
+          'badge-background-color',
+          badge.badgeBackgroundColor,
+        );
         merchCard.setAttribute('badge-color', badge.badgeColor);
         merchCard.setAttribute('badge-text', badge.badgeText);
       }
     }
-    if (actionMenuContent !== null) {
-      merchCard.setAttribute('action-menu', true);
-      merchCard.append(createTag('div', { slot: 'action-menu-content' }, actionMenuContent.innerHTML));
-    }
-    if (image !== undefined) {
-      const imageSlot = createTag('div', { slot: 'bg-image' });
-      imageSlot.appendChild(image);
-      merchCard.appendChild(imageSlot);
-    }
-    if (!icons || icons.length > 0) {
-      const iconImgs = Array.from(icons).map((icon) => {
-        const img = {
-          src: icon.querySelector('img').src,
-          alt: icon.querySelector('img').alt,
-        };
-        return img;
-      });
-      merchCard.setAttribute('icons', JSON.stringify(Array.from(iconImgs)));
-      icons.forEach((icon) => icon.remove());
-    }
-    parseContent(el, altCta, cardType, merchCard);
-    const footer = createTag('div', { slot: 'footer' });
-    if (ctas) {
-      decorateButtons(ctas);
-      footer.append(ctas);
-      merchCard.appendChild(footer);
-    }
-    if (styles.includes('secure')) {
-      replaceKey('secure-transaction', getConfig())
-        .then((key) => merchCard.setAttribute('secure-label', key));
-    }
-    if (altCta) {
-      const label = checkBoxLabel(ctas, altCta);
-      if (label !== null) {
-        merchCard.setAttribute('checkbox-label', label);
-      }
+  }
+
+  const actionMenuContent = cardType === 'catalog'
+    ? getActionMenuContent(el)
+    : null;
+  if (actionMenuContent) {
+    merchCard.setAttribute('action-menu', true);
+    merchCard.append(
+      createTag(
+        'div',
+        { slot: 'action-menu-content' },
+        actionMenuContent.innerHTML,
+      ),
+    );
+  }
+  const ctas = el.querySelector('p > strong a, p > em a')?.closest('p');
+  if (image !== undefined) {
+    const imageSlot = createTag('div', { slot: 'bg-image' });
+    imageSlot.appendChild(image);
+    merchCard.appendChild(imageSlot);
+  }
+  if (!icons || icons.length > 0) {
+    const iconImgs = Array.from(icons).map((icon) => {
+      const img = {
+        src: icon.querySelector('img').src,
+        alt: icon.querySelector('img').alt,
+      };
+      return img;
+    });
+    merchCard.setAttribute(
+      'icons',
+      JSON.stringify(Array.from(iconImgs)),
+    );
+    icons.forEach((icon) => icon.remove());
+  }
+  if (styles.includes('add-stock')) {
```

Comment:
Should this be inside a function? Kinda looks big to be inline inside init(), which is already big